### PR TITLE
Andrew7234/runtime tx denoms

### PR DESCRIPTION
**Pull request by Andrew7234**
Adds denomination fields for runtime transactions as `fee_symbol` and `amount_symbol`. 

Also updates `is_likely_native_token_xfer` flag to be more accurate, resolving https://app.clickup.com/t/8693vwwpz

### CODE REVIEW

File: analyzer/runtime/runtime.go [hunk 1]
```diff
@@ -65,16 +65,16 @@ func NewRuntimeAnalyzer(
...
-	return m.runtimeMetadata.Denominations[sdkConfig.NativeDenominationKey].Symbol
+func nativeTokenSymbol(runtimeMetadata *sdkConfig.ParaTime) string {
+	return runtimeMetadata.Denominations[sdkConfig.NativeDenominationKey].Symbol
 }
 
 // StringifyDenomination returns a string representation of the given denomination.
 // This is simply the denomination's symbol; notably, for the native denomination,
 // this is looked up from network config.
-func (m *processor) StringifyDenomination(d sdkTypes.Denomination) string {
+func StringifyDenomination(runtimeMetadata *sdkConfig.ParaTime, d sdkTypes.Denomination) string {
```
Comment (Andrew7234) [inline analyzer/runtime/runtime.go:75]: The alternative to this refactor would be to copy these methods to `extract.go`, which would leave us with duplicate `processor.StringifyDenomination` as well as a `StringifyDenomination` functions in the same package. Not sure which is cleaner/more idiomatic, open to suggestions.

**Review by pro-wh — APPROVED** [review]

File: storage/client/client.go [hunk 2]
```diff
@@ -1483,7 +1485,7 @@ func (c *StorageClient) RuntimeTransactions(ctx context.Context, p apiTypes.GetR
 		// TODO: Similarly to above, this application logic doesn't belong here (= the DB layer);
 		// move it out if we establish a separate app/logic layer.
 		if t.Method != nil {
-			if *t.Method == "accounts.Transfer" {
+			if *t.Method == "accounts.Transfer" && t.AmountSymbol != nil && *t.AmountSymbol == c.nativeTokenSymbol(runtimeFromCtx(ctx)) {
```
[the remark this answers is not in the document]
Comment (Andrew7234) [reply storage/client/client.go:1488]: Oh it's from an old ticket on the backlog, essentially we want to improve the heuristic used to set this flag: https://app.clickup.com/t/8693vwwpz

[the remark this answers is not in the document]
Comment (pro-wh) [reply storage/client/client.go:1488]: oh ok thanks

File: storage/migrations/01_runtimes.up.sql [hunk 3]
```diff
@@ -43,15 +43,19 @@ CREATE TABLE chain.runtime_transactions
...
 
   -- Transaction contents.
-  method      TEXT,         -- accounts.Transfer, consensus.Deposit, consensus.Withdraw, evm.Create, evm.Call. NULL for malformed and encrypted txs.
-  body        JSONB,        -- For EVM txs, the EVM method and args are encoded in here. NULL for malformed and encrypted txs.
-  "to"        oasis_addr,   -- Exact semantics depend on method. Extracted from body; for convenience only.
-  amount      UINT_NUMERIC, -- Exact semantics depend on method. Extracted from body; for convenience only.
+  method         TEXT,         -- accounts.Transfer, consensus.Deposit, consensus.Withdraw, evm.Create, evm.Call. NULL for malformed and encrypted txs.
+  body           JSONB,        -- For EVM txs, the EVM method and args are encoded in here. NULL for malformed and encrypted txs.
+  "to"           oasis_addr,   -- Exact semantics depend on method. Extracted from body; for convenience only.
+  amount         UINT_NUMERIC, -- Exact semantics depend on method. Extracted from body; for convenience only.
```
[the remark this answers is not in the document]
Comment (Andrew7234) [reply storage/migrations/01_runtimes.up.sql:56]: 🤨 good eye, no idea where that came from. fixed!

**Comment by Andrew7234** [comment]
Huh we still have damask emerald disabled in our e2e_regression tests. I tried again and still had errors trying to query historical balances in the runtime finalizeFastSync. The [comment](https://github.com/oasisprotocol/nexus/blob/main/tests/e2e_regression/damask/e2e_config_1.yml#L23) says we probably need to redeploy the archive node now that [5540](https://github.com/oasisprotocol/oasis-core/pull/5540) is merged, which seems to have happened. The archive node is on labs infra and is running this [image tag](https://hub.docker.com/layers/oasislabs/emerald-paratime/o22.2.12-e10.0.0-w3.4.0-c2.6.2-s0.6.4-k0.3.3/images/sha256-cd6e2061eead055a16340a90482c2d8a937dd219a0aba94f66b17cb4465b2d73?context=repo), which was built 5mo ago and should have worked. Will file a follow-up ticket to investigate further

**Pull request merged**